File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Btree/SearchTarget.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

namespace Db4objects.Db4o.Internal.Btree
{
	/// <exclude></exclude>
	public sealed class SearchTarget
	{
		public static readonly Db4objects.Db4o.Internal.Btree.SearchTarget Lowest = new Db4objects.Db4o.Internal.Btree.SearchTarget
			("Lowest");

		public static readonly Db4objects.Db4o.Internal.Btree.SearchTarget Any = new Db4objects.Db4o.Internal.Btree.SearchTarget
			("Any");

		public static readonly Db4objects.Db4o.Internal.Btree.SearchTarget Highest = new 
			Db4objects.Db4o.Internal.Btree.SearchTarget("Highest");

		private readonly string _target;

		public SearchTarget(string target)
		{
			_target = target;
		}

		public override string ToString()
		{
			return _target;
		}
	}
}

File: db4oj/core/src/com/db4o/internal/btree/SearchTarget.java
/* Copyright (C) 2006  Versant Inc.  http://www.db4o.com */

package com.db4o.internal.btree;

/**
 * @exclude
 */
public final class SearchTarget {
    
    public static final SearchTarget LOWEST = new SearchTarget("Lowest");
    
    public static final SearchTarget ANY = new SearchTarget("Any");
    
    public static final SearchTarget HIGHEST = new SearchTarget("Highest");
    
    private final String _target;

    
    public SearchTarget(String target) {
        _target = target;
    }

    public String toString() {
        return _target;
    }
    
}
